File: connativo.rb
require 'sinatra'
require 'humanize'

# Configurar el idioma global de la gema a español
Humanize.config.default_locale = :es

set :port, 4567

get '/connativo' do
  # Convertir el parámetro recibido en la URL a un número entero
  numero = params['n'].to_i

  # Ejecutar la conversión nativa
  numero.humanize
end

File: connativo.go
package main

import (
    "fmt"
    "net/http"
    "strconv"
    "moul.io/number-to-words"
)

func main() {
    http.HandleFunc("/connativo", func(w http.ResponseWriter, r *http.Request) {
        // Obtener y convertir el parámetro a entero
        nStr := r.URL.Query().Get("n")
        n, err := strconv.Atoi(nStr)
        if err != nil {
            n = 0
        }

        // Realizar la conversión nativa utilizando el idioma español (es-es)
        resultadoLetras := ntw.IntegerToEsEs(n)

        fmt.Fprint(w, resultadoLetras)
    })

    fmt.Println("Servidor de la Versión 3 corriendo en http://localhost:3000")
    http.ListenAndServe(":3000", nil)
}
